tools/functional-tester/etcd-tester/tester.go

```diff
@@ -80,45 +79,40 @@ func (tt *tester) runLoop() {
 		}
 	}
 
-	plog.Printf("%s functional-tester is finished", tt.logPrefix())
+	plog.Infof("%s functional-tester is finished", tt.logPrefix())
 }
 
-func (tt *tester) doRound(round int) (bool, error) {
+func (tt *tester) doRound(round int) error {
 	for j, f := range tt.failures {
 		caseTotalCounter.WithLabelValues(f.Desc()).Inc()
 		tt.status.setCase(j)
 
 		if err := tt.cluster.WaitHealth(); err != nil {
-			plog.Printf("%s wait full health error: %v", tt.logPrefix(), err)
-			return false, nil
+			return fmt.Errorf("wait full health error: %v", err)
 		}
-		plog.Printf("%s injecting failure %q", tt.logPrefix(), f.Desc())
+		plog.Infof("%s injecting failure %q", tt.logPrefix(), f.Desc())
 		if err := f.Inject(tt.cluster, round); err != nil {
-			plog.Printf("%s injection error: %v", tt.logPrefix(), err)
-			return false, nil
+			return fmt.Errorf("injection error: %v", err)
```

Comment:
I think we should reset the tester if any failure injection or recovery has failed because we don't really know the state of the system if those operation fail.  So resetting the system state through clean up will before next round of testing.